dss/operations/checkout.py

```diff
@@ -0,0 +1,188 @@
+        else:
+            raise RuntimeError(f'attempted to delete {bucket}/{key} but it did not work ;(')
+
+    def _verify_get(self, handle, bucket, key):
+        """ Helper Function to get file_metadata from cloud-blobstore"""
+        try:
+            file_metadata = json.loads(handle.get(bucket=bucket, key=key).decode('utf-8'))
+        except BlobNotFoundError:
+            logger.warning(f'unable to locate {bucket}/{key}')
+            return None
+        return file_metadata
+
+    def _verify_blob_existance(self, handle, bucket, key):
+        """Helper Function to see if blob data exists"""
+        try:
+            handle.get(bucket=bucket, key=key)
+        except BlobNotFoundError:
+            return False
+        return True
+
+    def _sleepy_checkout(self, function=None, **kwargs):
+        status = False
+        token = None
+        key = kwargs["file_metadata"]['name'] if kwargs.get('file_metadata') else kwargs.get("bundle_uuid")
+        logger.warning(f'starting {function.__name__} on {key}')
+        while status is not True:
+            kwargs['token'] = token
+            token, status = function(**kwargs)
+            if not status:
+                time.sleep(4)  # wait 4 seconds for checkout to complete, then try again
+
+    def _parse_key(self, key):
+        try:
+            version = VERSION_REGEX.search(key).group(0)
+            uuid = UUID_REGEX.search(key).group(0)
+        except IndexError:
+            RuntimeError(f'unable to parse the key {key}')
+        return uuid, version
+
+    def process_keys(self):
+        raise NotImplementedError()
+
+    def __call__(self, argv: typing.List[str], args: argparse.Namespace):
+        if self.keys is not None:
+            self.process_keys()
+
+
+checkout = dispatch.target("checkout", help=__doc__)
+
+
+@checkout.action("remove",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                            "--keys": dict(nargs="+", help="keys to remove from checkout", required=True)})
+class remove(CheckoutHandler):
+    def process_keys(self):
+        """
+        Remove a bundle from the checkout bucket
+        """
+        for _key in self.keys:
+            if DSS_BUNDLE_KEY_REGEX.match(_key):
+                for key in self.handle.list(self.bucket, _key):  # handles checkout/bundle/*
+                    self._verify_delete(self.handle, self.bucket, key)
+                uuid, version = self._parse_key(_key)
+                manifest = get_bundle_manifest(replica=self.replica, uuid=uuid, version=version)
+                if manifest is None:
+                    logger.warning(f"unable to locate manifest for fqid: {self.bucket}/{_key}")
+                    continue
+                for _files in manifest['files']:
+                    key = compose_blob_key(_files)
+                    self._verify_delete(self.handle, self.bucket, key)
+            elif FILE_PREFIX in _key:
```

Comment:
Should be elif _key.startswith(FILE_PREFIX). Odds are low that "files" might be a part of the uuid, but it's not impossible.